### ISSUE TITLE
Database conversion fails 14.02 to 15.XX

### ISSUE THREAD

**Issue by LdrAms** [MicrosoftDocs/dynamics365smb-devitpro-pb#999]
Hi,

I'm trying to upgrade a 14.02 DB to a 15.02. Everything goes fine (without errors) until i try to open the web client, then the the following error occurrs (translated from Spanish):

Type: Microsoft.Dynamics.Nav.Types.Exceptions.NavSqlException
ErrorNumber: 1038
...
Alias '' or [] are not allowed. ...
  SQL:
  IF NOT EXISTS (SELECT * FROM sys.columns WHERE object_id = OBJECT_ID(N'') AND name = N'$systemId')BEGIN
  ALTER TABLE [] ADD [$systemId] [uniqueidentifier] NOT NULL CONSTRAINT "MDF$$$systemId" DEFAULT NEWSEQUENTIALID() CONSTRAINT "$$systemId" UNIQUE NONCLUSTERED ("$systemId")
  END

The new systemId column was not created at any table. Any idea of what is going wrong?

Thanks,

**Comment by LdrAms** [comment]
In fact, the process fails at Task 8 [Synchronize the tenant] with the following error:
Sync-NAVTenant : The following SQL error was not expected:
An object or column name is missing or empty. Check if all columns of the SELECT INTO instructions have a name. For other instructions, look for empty alias names. Aliases defined as '' or [] are not allowed. Change the alias to a valid name.
On line: 1 Character: 1

**Comment by LdrAms** [comment]
For it may help, each time I restart the service, the followinf errors arise at the event source:
The package ID 'e46ac93e-f594-4408-a1af-7367e0b73369' for the tenant 'default' does not correspond to an installed app.
The package ID '2ab7733d-e97e-4c74-b4ae-cb341b865e0c' for the tenant 'default' does not correspond to an installed app.
If I try to install the base app published extension, the followig error occurs:
Install-NAVApp : The XXXX database has version 1250 which is not supported by this version of Microsoft Dynamics 365 Business Central Server. Please contact your administrator.

**Comment by jswymer** [comment]
Hello,

Just to verify, because the article link is mssing. Are you follwoing the steps in [Upgrading an Unmodified Application to Dynamics 365 Business Central 2019 Release Wave 2](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/upgrade/upgrade-unmodified-application)? 

I have not seen this issue before. You could run the following SQL query against the Object Metadata Snapshot table, and see whether there are any empty IDs or Names.

SELECT [Object ID], [Name], [Data Per Company] FROM [Object Metadata Snapshot] WHERE [Object Type] = 1

**Comment by LdrAms** [comment]
Hi,
Yes, I'm following the steps. And yes, there are some rows with empty names:
Object ID
6701		0
6703		0
6704		0
6705		0
6707		0
I've delete those rows and I was able to execute Sync-NAVTenant (Task 8) without errors. I suppose that it was from a previous upgrade.

**Comment by LdrAms** [comment]
Sync-NavApp with System application works fine also but, with “Base Application” another error arise (translated):
The following SQL error was unexpected:
Parameter @objname is ambiguous or the value @objtype (INDEX) is wrong.
Warning: changing  any part of an object name, the scripts stored procedures may stop working. (same warning 9 times)
At the event viewer the statement causing the error is:

SQL statement:
  exec sp_rename N'"dbo"."Company name$G_L Account$0"', N'Company name$G_L Account$437dbf0e-84ff-417a-965d-ed2bb9650972$Key1'exec sp_rename N'"dbo"."COMPANY NAME$G_L Account"."$1"', N'$Key2', N'INDEX'exec sp_rename N'"dbo"."COMPANY NAME$G_L Account"."$2"', N'$Key3', N'INDEX'exec sp_rename N'"dbo"."COMPANY NAME$G_L Account"."$3"', N'$Key4', N'INDEX'exec sp_rename N'"dbo"."COMPANY NAME$G_L Account"."$4"', N'$Key5', N'INDEX'exec sp_rename N'"dbo"."COMPANY NAME$G_L Account"."$7"', N'$Key8', N'INDEX'exec sp_rename N'"dbo"."COMPANY NAME$G_L Account"."$8"', N'$Key9', N'INDEX'exec sp_rename N'"dbo"."COMPANY NAME$G_L Account"."$9"', N'$Key10', N'INDEX'exec sp_rename N'"dbo"."COMPANY NAME$G_L Account$$systemId"', N'COMPANY NAME$G_L Account$437dbf0e-84ff-417a-965d-ed2bb9650972$$systemId'; exec sp_rename N'"dbo"."COMPANY NAME$G_L Account"', N'COMPANY NAME$G_L Account$437dbf0e-84ff-417a-965d-ed2bb9650972'IF NOT EXISTS (SELECT * FROM sys.columns WHERE object_id = OBJECT_ID(N'COMPANY NAME$G_L Account$437dbf0e-84ff-417a-965d-ed2bb9650972') AND name = N'$systemId')BEGIN
  ALTER TABLE [COMPANY NAME$G_L Account$437dbf0e-84ff-417a-965d-ed2bb9650972] ADD [$systemId] [uniqueidentifier] NOT NULL CONSTRAINT "MDF$COMPANY NAME$G_L Account$437dbf0e-84ff-417a-965d-ed2bb9650972$$systemId" DEFAULT NEWSEQUENTIALID() CONSTRAINT "COMPANY NAME$G_L Account$437dbf0e-84ff-417a-965d-ed2bb9650972$$systemId" UNIQUE NONCLUSTERED ("$systemId")
  END

**Comment by jswymer** [comment]
Hello again, Sorry, but I am not sure what is causing the error. Especially if there are no code customizations in the application. What build number are you are using, is it a language (localized) or W1 build? Perhaps the best is to create a support ticket.

**Comment by LdrAms** [comment]
Hi, we are using build 15.0.38951 and a Spanish localized version. About the support ticket, do you mean at the partnersource?

Thanks any way.

**Comment by jswymer** [comment]
Hello, I tried this out on a ES build 14.0.2 to 15.0.3 with an unmodified base application, and it worked. 
I did not know much about the support channel, unfortunalely. But if you are a customer, then you should contact your partner. If you are partner, then you should contact your Microsoft Support engineer or any local Microsoft Support office or site.  I hope this helps.

**Comment by LdrAms** [comment]
Hello,
We are a partner, we have done it several times with another databases and it works fine, it's this particular database that have something wrong, probabily from a previous (years ago) upgrade. We'll try a support ticket.

-- Alejandro --

**Comment by jswymer** [comment]
Hello, I hope it works out through support. If it is a documentation issue, it should filter down to us. So for now I will close this issue. But if you find out anything that is related to the documentation, feel free to add it to this issue and we will gladly reopen it. Thanks!

**Comment by LdrAms** [comment]
Hi,

I finally find out what was going on, so for it may help someone else, here is what was happening. From a previous upgrade, some tables remains with wrong indexes created at sql server in certain tables, G_L Account was the first, if you, from C/Side disable ALL indexes except primary key, and have a look at Sql Server you will find that some indexes remains there, I then remove them from SQL Server directly on each G_L Account table corresponding to each company at the database so there was only the $0. Then, from C/Side again, I enabled all the indexes that was previously enabled, and that’s all. I had to do the same with several tables with the same problem but then, everything went fine.

Hope it helps.

-- Alejandro --
